src/EscapeAnalysis.jl

```diff
@@ -165,29 +411,66 @@ Extended lattice that maps arguments and SSA values to escape information repres
 - `state.arguments::Vector{EscapeLattice}`: escape information about "arguments" – note that
   "argument" can include both call arguments and slots appearing in analysis frame
 - `ssavalues::Vector{EscapeLattice}`: escape information about each SSA value
+- `aliaset::IntDisjointSet{Int}`: a disjoint set that represents aliased arguments and SSA values
 """
 struct EscapeState
     arguments::Vector{EscapeLattice}
     ssavalues::Vector{EscapeLattice}
+    aliasset::IntDisjointSet{Int}
```

Comment:
I don't quite understand `aliasset`, or rather, the difference between the information stored in `aliasset` and that stored in `FieldSets`. Do we use `aliasset` for performance reasons? IUIC we could (albeit with probably slow performance) write `get_aliases(state::EscapeState, x, ir)` by searching for `x`'s value within the `FieldSets` of `state.arguments` and/or `state.ssavalues`?

Reply:
I think I introduced `aliasset` so that we can track a case like below:
```julia
analyze_escapes((String,)) do a # => ReturnEscape
    o1  = Ref(a)                # => NoEscape
    o2  = Ref(o1)               # => NoEscape
    o1′ = o2[]                  # (aliased to `o1`)
    a′  = o1′[]                 # (needs to propagate ReturnEscape to `a`)
    return a′
end
```

That is, `FieldSets` tracks values of a field that are known from `Expr(:new, ...)` or `setfield!` call, 
but in order to handle a case where escape information propagates through values that are aliased by `getfield` chain,
we need some external set that holds all the possible values that can be aliased to a field in interest.

Having said that, now I think the design of `FieldSets` is wrong.
I made `FieldSets` hold actual values (i.e. `SSAValue`/`Argument`) and escape information is propagated by tracking those values (and aliased values by looking at `aliasset`), just because I had SROA in my mind as a downstream optimization that can directly use that information.
But the important observation here is that field information is something like type information, and it is something better propagated in forward direction. This PR tries to analyze field information in a backward analysis, but it leads a complex and inefficient (and likely wrong) data flow.
Clearly, it may be better not to include a design that is specific downstream optimization in mind (like SROA), and I think we can have a better escape analysis implementation by making `FieldSets` somewhat like `aliasset`, and just trying to propagate escape information without holding field values.